### PR TITLE
[MRG+1] Remove support for Python 3.7

### PR DESCRIPTION
**Pull request by aaronreidsmith**
# Description

Python 3.7 is due to be [end-of-lifed on June 27, 2023](https://endoflife.date/python). This PR preps us to remove support for it.

Think this will depend on https://github.com/alkaline-ml/pmdarima-docker/pull/26

## Type of change

- [X] Python version removal

# How Has This Been Tested?

3.7 tests no longer exist, everything else still works

# Checklist:

- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: .github/workflows/build_and_deploy.yml [hunk 1]
```diff
@@ -20,7 +20,7 @@ jobs:
   build-and-deploy:
     strategy:
       matrix:
-        python3-minor-version: [7, 8, 9, 10, 11, 12]
+        python3-minor-version: [9, 10, 11]
```
Comment (aaronreidsmith) [inline .github/workflows/build_and_deploy.yml:23]: Think this was a bad conflict resolution. We support 3.12 today

File: doc/whats_new.rst [hunk 2]
```diff
@@ -11,6 +11,8 @@ v0.8.1) will document the latest features.
 -------------------------------------------------
 
 * Pin numpy to ``<2.0.0`` while work is done to support numpy ``>=2.0.0``
+* Remove support for Python 3.7 (end-of-life 2023-06-23)
+* Remove support for Python 3.8 (end-of-life 2024-10-07)
```
Comment (aaronreidsmith) [inline doc/whats_new.rst:15]: pmdarima 2.0.4 is already released, these should go under `2.0.5`

File: setup.py [hunk 3]
```diff
@@ -239,7 +239,6 @@ def do_setup():
                         'Operating System :: Unix',
                         'Operating System :: MacOS',
                         'Programming Language :: Python :: 3',
-                        'Programming Language :: Python :: 3.7',
                         'Programming Language :: Python :: 3.8',
```
Comment (aaronreidsmith) [inline setup.py:242]: This should go too

File: .github/workflows/build_and_deploy.yml (hunk 1 above)
Comment (tgsmith61591) [reply .github/workflows/build_and_deploy.yml:23]: Good catch

File: doc/whats_new.rst (hunk 2 above)
Comment (tgsmith61591) [reply doc/whats_new.rst:15]: Also good catch. Conflict resolution :)

**Review by tgsmith61591 — APPROVED** [review]

**Pull request merged**